File: r/words/test/numerical_test.R
library(testthat)

source("../src/numerical.R")

test_that("fibonacci", {
  expect_equal(fibonacci(-3), -1)
  expect_equal(fibonacci(10), 55)
  expect_equal(fibonacci(15), 610)
})

test_that("factorial", {
  expect_equal(factorial(-3), -1)
  expect_equal(factorial(5), 120)
  expect_equal(factorial(10), 3628800)
})

test_that("greatest common divisor", {
  expect_equal(gcd(48, 18), 6)
  expect_equal(gcd(100, 25), 25)
  expect_equal(gcd(17, 13), 1)
})

test_that("least common multiple", {
  expect_equal(lcm(4, 6), 12)
  expect_equal(lcm(21, 6), 42)
  expect_equal(lcm(7, 3), 21)
})

File: typescript/words/test/numerical.test.ts
import { fibonacci, factorial, gcd, lcm } from '../src/numerical';

describe ('Numerical Functions', () => {
  test('fibonacci', () => {
    expect(fibonacci(-3)).toBe(-1);
		expect(fibonacci(10)).toBe(55);
		expect(fibonacci(15)).toBe(610);
  });
	test('factorial', () => {
		expect(factorial(-3)).toBe(-1);
		expect(factorial(5)).toBe(120);
		expect(factorial(10)).toBe(3628800);
	});
	test('greatest common divisor', () => {
		expect(gcd(48, 18)).toBe(6);
		expect(gcd(100, 25)).toBe(25);
		expect(gcd(17, 13)).toBe(1);
	});
	test('least common multiple', () => {
		expect(lcm(4, 6)).toBe(12);
		expect(lcm(21, 6)).toBe(42);
		expect(lcm(2, 3)).toBe(6);
	});
});
